src/db.c

```diff
@@ -880,6 +885,7 @@ int flushCommandCommon(client *c, int type, int flags, SlotsFlush *sflush) {
         elapsedStart(&c->bstate.lazyfreeStartTime);
 
         c->bstate.timeout = 0;
+        c->flags |= CLIENT_PENDING_COMMAND;
```

Comment:
are you saying that we were missing n update of the replication offset? can you describe the flow?

Reply:
No, the original flow is fine, I just want to use it to illustrate the following flow that we continue with the previous command, so we can assume that we will continue to process it after `unblockclient()`.

Reply:
so this is just a cleanup not to to avoid unblockclient reset a command that is still being handled?
and has no actual implications? maybe you can improve the description..

Reply:
i'll change it. 😅

Reply:
updated the top comment.

Reply:
doesn't it still makes it look like there was a bug with the replication offset?

Reply:
😏 i think it's okay this time.